spec/state_machines/workflows/predicates_spec.rb

```diff
@@ -0,0 +1,147 @@
+          responder: @assigned_responder
+        ),
+        case_drafting_flagged_press: create(
+          :case_being_drafted,
+          :flagged,
+          :press_office,
+          approver: @press_officer,
+          responder: @assigned_responder
+        ),
+        case_drafting_trigger: create(
+          :case_being_drafted,
+          :flagged_accepted,
+          approver: @disclosure_specialist,
+          approving_team: @team_disclosure,
+          responder: @assigned_responder
+        ),
+        case_drafting_trigger_press: create(
+          :case_being_drafted,
+          :flagged_accepted,
+          :press_office,
+          approver: @press_officer,
+          disclosure_specialist: @disclosure_specialist,
+          responder: @assigned_responder
+        ),
+        case_unassigned: create(
+          :case
+        ),
+        case_unassigned_flagged: create(
+          :case,
+          :flagged,
+          :dacu_disclosure
+        ),
+        case_unassigned_flagged_press: create(
+          :case,
+          :flagged,
+          :press_office,
+          approver: @press_officer,
+          responder: @assigned_responder
+        ),
+        case_unassigned_trigger: create(
+          :case,
+          :flagged_accepted,
+          :dacu_disclosure
+        ),
+        case_unassigned_trigger_press: create(
+          :case,
+          :flagged_accepted,
+          :press_office,
+          approver: @press_officer,
+          disclosure_specialist: @disclosure_specialist,
+          responder: @assigned_responder
+        ),
+      }
+    end
+
+    def all_users
+      # Users used by the permit_only_these_combinations matcher in combination
+      # with all_cases().
+      {
+        assigned_responder:             @assigned_responder,
+        another_responder:              @another_responder,
+        disclosure_bmt_user:            @disclosure_bmt_user,
+        disclosure_specialist:          @disclosure_specialist,
+        disclosure_specialist_coworker: @disclosure_specialist,
+        press_officer:                  @press_officer,
+      }
+    end
+
+    def all_cases
+      @all_cases
+    end
+
+    describe :responder_is_member_of_assigned_team? do
+      it {
+        # This matcher will expect the given predicate in the spec description
+        # to only allow the combinations of [user, case] provided here. All
+        # other combinations of the case types in all_cases() and the user
+        # types in all_users() will be expected to fail with this predicate.
+        should permit_only_these_combinations(
+                 [:assigned_responder, :case_drafting],
+                 [:assigned_responder, :case_drafting_flagged],
+                 [:assigned_responder, :case_drafting_flagged_press],
+                 [:assigned_responder, :case_drafting_trigger],
+                 [:assigned_responder, :case_drafting_trigger_press],
+               )
+      }
```

Comment:
I realised over the weekend that what might be a nice pattern here is to do something like:

```
should give_permission_on(case_drafting).to(assigned_responder)
```

For tests where more than one user is permitted to do something, we could have something like:

```
describe :reassign_user do
  it {
    should give_permission_on(case_drafting).to assigned_responder,
                                                coworker_responder
  }
end
```

It's just an idea and, admittedly, while it may make more sense to group by case, now that I've sketched it out I kind of like the way the existing implementation looks. It seems that it might be easier to read. Thoughts welcome from those who have read thus far.